### PR TITLE
UNR-3524 Make the Launch Configuration Editor window modal so as to avoid multi instances

### PR DESCRIPTION
**Pull request by yujunting**
#### Description
Make the Launch Configuration Editor window modal so as to avoid multi instances

#### Release note
REQUIRED: clean needless code about Launch Configuration Editor, remove class UTransientUObjectEditor, and rename LaunchConfigEditor.h&.cpp to LaunchConfigurationEditor.h&.cpp to make the file name same as the class name.

this issue only fixes a simple bug. so I think it doesn't need to modify the changelog.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp [hunk 1]
```diff
@@ -3,25 +3,14 @@
...
-	if (MyLayout != nullptr)
-	{
-		TArray<TWeakObjectPtr<UObject>> Objects;
-		MyLayout->GetObjectsBeingCustomized(Objects);
-		ULaunchConfigurationEditor* Editor = Objects.Num() > 0 ? Cast<ULaunchConfigurationEditor>(Objects[0].Get()) : nullptr;
-		MyLayout->ForceRefreshDetails();
-	}
-}
-
 void FLaunchConfigEditorLayoutDetails::CustomizeDetails(IDetailLayoutBuilder& DetailBuilder)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp:14]: It looks like this customization doesn't actually do anything. We should just remove this class.

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp [hunk 2]
```diff
@@ -1,23 +1,62 @@
...
 #include "PropertyEditor/Public/PropertyEditorModule.h"
-
-#include "Framework/Application/SlateApplication.h"
+#include "SpatialGDKSettings.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialRuntimeLoadBalancingStrategies.h"
 #include "Widgets/Input/SButton.h"
 #include "Widgets/Layout/SBorder.h"
 
+TSharedPtr<ULaunchConfigurationEditor> ULaunchConfigurationEditor::_instance = nullptr;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp:16]: Is this being used? Please remove if not.

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp [hunk 3]
```diff
@@ -1,23 +1,62 @@
...
+#include "IDesktopPlatform.h"
 #include "MainFrame/Public/Interfaces/IMainFrameModule.h"
 #include "PropertyEditor/Public/PropertyEditorModule.h"
-
-#include "Framework/Application/SlateApplication.h"
+#include "SpatialGDKSettings.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialRuntimeLoadBalancingStrategies.h"
 #include "Widgets/Input/SButton.h"
 #include "Widgets/Layout/SBorder.h"
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp:14]:
```suggestion
#include "Widgets/Input/SButton.h"
#include "Widgets/Layout/SBorder.h"

#include "SpatialGDKSettings.h"
#include "SpatialGDKDefaultLaunchConfigGenerator.h"
#include "SpatialRuntimeLoadBalancingStrategies.h"
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp (hunk 1 above)
Comment (yujunting) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp:14]: it's used here. but I'm not sure if it really needed. 
PropertyModule.RegisterCustomClassLayout(ULaunchConfigurationEditor::StaticClass()->GetFName(), FOnGetDetailCustomizationInstance::CreateStatic(&FLaunchConfigEditorLayoutDetails::MakeInstance));

If not, I can remove it.

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp:14]: Yes, I think it's not needed. I tried removing that line and the behavior is not affected. I believe `IDetailCustomization` is usually used to add elements to the settings section, but here nothing is actually being added.

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp [hunk 4]
```diff
@@ -113,48 +140,42 @@ UTransientUObjectEditor* UTransientUObjectEditor::LaunchTransientUObjectEditor(c
...
-		FSlateApplication::Get().AddWindowAsNativeChild(NewSlateWindow, ParentWindow.ToSharedRef());
+		FSlateApplication::Get().AddModalWindow(NewSlateWindow, InParentWindow.ToSharedRef());
 	}
 	else
 	{
-		FSlateApplication::Get().AddWindow(NewSlateWindow);
+		FSlateApplication::Get().AddModalWindow(NewSlateWindow, nullptr);
 	}
-
-	NewSlateWindow->RegisterActiveTimer(0.5, FWidgetActiveTimerDelegate::CreateLambda([NewSlateWindow](double, float)
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp:153]: I acknowledge this was a hack to get the right size for the windows and circumvent Slate's layout logic, but why did it need to be removed ?

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp (hunk 1 above)
Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigEditorLayoutDetails.cpp:14]: The customization was needed until the ServerWorkerMap was removed from the settings. You can remove it now.

File: SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp (hunk 4 above)
Comment (yujunting) [reply SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp:153]: It resizes the window to the desired size after opened 0.5s. If changed to a modal window, this code is blocked until closing the window.
Give the windows a initialize size is simple to fix that.

**Comment by improbable-valentyn** [comment]
@ImprobableNic are there any other object editors currently planned? If not, I'd suggest we go ahead with removing TransientUObjectEditor for now and add it back once we have more than one use case.

**Pull request merged**